libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -1628,22 +1626,22 @@ void BeaconSetupService::SetTimeToProceed(BeaconSetupService::State state)
     // If not ahead in time, the DKG must have failed before. Algorithmically, and importantly
     // deterministically, decide how long to increase the allotted DKG time (increment each time
     // by 1.5x to a maximum of MAX_DKG_MULTIPLIER)
-    expected_dkg_timespan_ = time_per_iteration;
+    expected_dkg_timespan_ = uint64_t(time_per_state * time_slots_in_dkg_);
     uint16_t failures      = 0;
 
     while (reference_timepoint_ < current_time)
     {
       failures++;
       reference_timepoint_ += expected_dkg_timespan_;
       expected_dkg_timespan_ += std::min(expected_dkg_timespan_ + (expected_dkg_timespan_ / 2),
-                                         time_per_iteration * MAX_DKG_BOUND_MULTIPLE);
+                                         time_per_state * MAX_DKG_BOUND_MULTIPLE);
     }
```

Comment:
In order for the time per slot to increase in each iteration you need to change time_per_state here so that it gets multiplied by 1.5 with each failure, no?

Reply:
OK, I have tested it and the logic is (now) correct:

```
    while (reference_timepoint_ < current_time)
    {
      failures++;
      reference_timepoint_ += expected_dkg_timespan_;
      expected_dkg_timespan_ = std::min(
          (expected_dkg_timespan_ + (expected_dkg_timespan_ / 2)),
          static_cast<uint64_t>(time_per_state * time_slots_in_dkg_ * MAX_DKG_BOUND_MULTIPLE));
    }

```

Each failure adds itself + 1/2, so 1.5* each time to a multiple of 4.